sqlmesh/core/snapshot/evaluator.py

```diff
@@ -175,7 +177,18 @@ def apply(query_or_df: QueryOrDF, index: int = 0) -> None:
                                 execute(exp.select(existing_limit.expression)).rows[0][0],
                             )
                     return query_or_df.head(limit) if hasattr(query_or_df, "head") else self.adapter._fetch_native_df(query_or_df.limit(limit))  # type: ignore
-
+                elif (
+                    self.adapter.SUPPORTS_INSERT_OVERWRITE
```

Comment:
just to confirm, engines that don't support insert overwrite are not affected because they do a DELET FROM statement with start and end. wheras support insert overwrite uses dynamic partitioning.

Reply:
Yes the DELETE FROM combined with the append on future dataframes is why it works.